File: 20250613_global_logic/gl_010_yaml_groovy/012_typescript/src/fizz-buzz.ts
/**
 * Fizz Buzz app.
 */

/**
 * fizz_buzz function.
 */
export function fizzBuzz(n: number): string {
  let result = '';
  if (n % 3 === 0) {
    result += 'Fizz';
  }
  if (n % 5 === 0) {
    result += 'Buzz';
  }
  return result || n.toString();
}

/**
 * Main function to run fizz_buzz for 1-100.
 */
export function main(): void {
  for (let i = 1; i <= 100; i++) {
    console.log(fizzBuzz(i));
  }
}

// Run when executed directly (ES module check)
if (process.argv[1] && process.argv[1].endsWith('fizz-buzz.ts')) {
  main();
}

File: 20250613_global_logic/gl_010_yaml_groovy/011_python/src/gl_011/fizz_buzz.py
"""Fizz Buzz app."""


def fizz_buzz(n: int) -> str:
    """fizz_buzz function."""
    result = ""
    if n % 3 == 0:
        result += "Fizz"
    if n % 5 == 0:
        result += "Buzz"
    return result or str(n)


def main():
    """Main function to run fizz_buzz for 1-100."""
    for i in range(1, 101):
        print(fizz_buzz(i))


if __name__ == "__main__":
    main()
